### PR TITLE
[Core][Tiny]Shorter thread name

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
In linux the thread name could not be longer than 15 chars.
When we use command like top, we are easy being confused by similar thread name like `resource_report_poller` and `resource_report_broadcaster` because they are both show `resource_report`.

This pr uses abbr to make the thread names shorter.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/gcs_server_address_updater.cc [hunk 1]
```diff
@@ -27,7 +27,7 @@ GcsServerAddressUpdater::GcsServerAddressUpdater(
       update_func_(update_func),
       updater_runner_(updater_io_service_),
       updater_thread_([this] {
-        SetThreadName("gcs_address_updater");
+        SetThreadName("gcs_address_upd");
```
Comment (suquark) [inline src/ray/core_worker/gcs_server_address_updater.cc:30]: "gcs_addr_updater"

File: src/ray/gcs/gcs_server/ray_syncer.h [hunk 2]
```diff
@@ -53,7 +53,7 @@ class RaySyncer {
   void Start() {
     poller_->Start();
     broadcast_thread_ = std::make_unique<std::thread>([this]() {
-      SetThreadName("resource_report_broadcaster");
+      SetThreadName("resource_brdcst");
```
Comment (suquark) [inline src/ray/gcs/gcs_server/ray_syncer.h:56]: broadcast -> bcast is more common, e.g., https://www.mpich.org/static/docs/v3.1/www3/MPI_Bcast.html

File: src/ray/core_worker/gcs_server_address_updater.cc (hunk 1 above)
Comment (WangTaoTheTonic) [reply src/ray/core_worker/gcs_server_address_updater.cc]: It's still one character exceed, but not a big problem.

File: src/ray/gcs/gcs_server/ray_syncer.h (hunk 2 above)
Comment (WangTaoTheTonic) [reply src/ray/gcs/gcs_server/ray_syncer.h]: got it.

**Review by suquark — APPROVED** [review]
LGTM. CI error seems not related.

**Review by jovany-wang — APPROVED** [review]

**Pull request merged**